### PR TITLE
Boefje scanning frequency support in scheduler

### PR DESCRIPTION
**Pull request by jpbruinsslot**
### Pre-Review

Since this PR's is blocked by changes to the katalogus (see https://github.com/minvws/nl-kat-coordination/issues/3065) I'm setting this PR's in review to gather feedback on the changes that have been made, and that expectations are met. Please note the `TODO` in the code on the parts where those changes of the katalogus are expected and need to be implemented.

### Changes

Based on the event of new or enabled boefjes a schedule is created when that boefje has a specified interval of execution (this is supplied by the `boefje.json`).

#### Before
 In the `BoefjeScheduler` before we would only check for enabled and new boefjes. For an enabled boefje the scheduler will gather all the OOI's this boefje can run on and will create `BoefjeTask` objects for this (`boefje * ooi = boefje tasks`). 

#### After

Now in addition to this we will check for enabled boefjes (`check_new_boefjes`) and create a `Schedule` for any boefje that has a interval specified.

### Issue link

_You have to create an issue to link to this PR. If this really is not possible, write a very detailed description here and add this PR to the project board directly._

_Please add the link to the issue after "Closes"._

Closes ...

### Demo

_Please add some proof in the form of screenshots or screen recordings to show (off) new functionality, if there are interesting new features for end-users._

### QA notes

_Please add some information for QA on how to test the newly created code._

---

### Code Checklist

- [ ] All the commits in this PR are properly PGP-signed and verified.
- [ ] This PR only contains functionality relevant to the issue.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] I have checked the documentation and made changes where necessary.
- [ ] I have performed a self-review of my code and refactored it to the best of my abilities.

- [ ] Tickets have been created for newly discovered issues.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Review by ammar92 — COMMENTED** [review]
Just a few remarks; skeleton looks good

File: mula/scheduler/schedulers/boefje.py [hunk 1]
```diff
@@ -312,6 +327,54 @@ def push_tasks_for_new_boefjes(self) -> None:
...
+            try:
+                self.ctx.datastores.schedule_store.create_schedule(schedule)
+            except storage.errors.StorageError as exc_db:
+                self.logger.error(
+                    "Could not create schedule for boefje %s",
+                    boefje.id,
+                    boefje_id=boefje.id,
+                    organisation_id=self.organisation.id,
+                    scheduler_id=self.scheduler_id,
+                    exc_info=exc_db,
```
[the remark this answers is not in the document]
Comment (jpbruinsslot) [reply mula/scheduler/schedulers/boefje.py:366]: nice one, thanks!

**Comment by jpbruinsslot** [comment]
This PR is at the moment blocked because of https://github.com/minvws/nl-kat-coordination/issues/3065 and the implementation of the manifest for a boefje

**Comment by jpbruinsslot** [comment]
This PR is put on hold and precedence is given to: https://github.com/minvws/nl-kat-coordination/pull/3529

This PR allows for an interval/schedule to be defined for a boefje that runs without an ooi to specifically run on. Meaning the `BoefjeTask` only references a Boefje not an input ooi. Meaning the task runner makes the decision on what to execute this boefje on.

**Comment by underdarknl** [comment]
Is this PR still relevant, or is there still a remaining function requested, but not implemented?

**Pull request closed without merging**